### PR TITLE
GCS:Input: Add warnings for bad reprojection settings

### LINKED ISSUE #961: UI needs to enforce that choices for reprojection are OK
as described in https://github.com/d-ronin/dRonin/pull/956#issuecomment-216092075

### PR DESCRIPTION
**Pull request by tracernz**
untitled

Fixes #961 
Although it doesn't enforce, just warns.

### CODE REVIEW

**Comment by tracernz** [comment]
Oops, changed default tab, force-pushed.

File: ground/gcs/src/plugins/config/configinputwidget.cpp [hunk 1]
```diff
@@ -1784,3 +1774,64 @@ void ConfigInputWidget::checkHangtimeConfig()
 
     m_config->lblHangTimeWarning->setVisible(warn);
 }
+
+void ConfigInputWidget::clearMessages(QWidget *widget, const QString type)
```
Comment (tracernz) [inline ground/gcs/src/plugins/config/configinputwidget.cpp:1778]: Should probs move these message functions to ConfigTaskWidget so they can be used elsewhere, and add Doxygen docs.

File: ground/gcs/src/plugins/config/configinputwidget.cpp [hunk 2]
```diff
@@ -1784,3 +1774,64 @@ void ConfigInputWidget::checkHangtimeConfig()
...
+}
+
+void ConfigInputWidget::addMessage(QWidget *widget, const QString type, const QString msg)
+{
+    QLabel *lbl = findChild<QLabel *>("lblMessage" + type);
+    if (!lbl) {
+        lbl = new QLabel(this);
+        lbl->setObjectName("lblMessage" + type);
+        lbl->setWordWrap(true);
+        widget->layout()->addWidget(lbl);
```
Comment (tracernz) [inline ground/gcs/src/plugins/config/configinputwidget.cpp:1794]: Should probably not assume the widget has a layout, rather add one if it doesn't before using it.

**Comment by mluessi** [comment]
Looks good. Did a quick test. Maybe it would make sense to highlight it more to alert the user of the error. I think many users will overlook/ignore the messages.

**Comment by mlyle** [comment]
LGTM

**Comment by mlyle** [comment]
Can improve usability later-- it's at least there.

**Pull request merged**